language/move-prover/tests/sources/functional/loops.move

```diff
@@ -0,0 +1,86 @@
+module VerifyLoops {
+
+    spec module {
+        pragma verify=false;
+    }
+
+    // ----------------------
+    // `aborts_if` statements
+    // ----------------------
+
+    public fun iter10_missing_inc_spec1() {
+        let i = 0;
+        while (i <= 10) { // an infinite loop
+            if (i > 10) abort 10;
+        }
+    }
+    spec fun iter10_missing_inc_spec1 { // TODO: What is the expected output for this spec?
+        aborts_if false;
+    }
+
+    public fun iter10_missing_inc_spec2() {
+        let i = 0;
+        while (i <= 10) { // an infinite loop
+            if (i > 10) abort 10;
+        }
+    }
+    spec fun iter10_missing_inc_spec2 { // TODO: What is the expected output for this spec?
+        aborts_if true;
+    }
+
+    public fun iter10_no_abort() {
+        let i = 0;
+        while ({
+            spec { assert i <= 11; };
+            (i <= 10)
+        }) {
+            if (i > 10) abort 10;
```

Comment:
Are you able to prove "assert false" just before abort 10? That is, make the conditional inside the loop body like this:

if (i > 10) {
    spec { assert false;}; abort 10
}

Reply:
Yes, I just check that the following is verified (Prover finds no errors):
```
    public fun iter10_no_abort() {
        let i = 0;
        while ({
            spec { assert i <= 11; };
            (i <= 10)
        }) {
            if (i > 10) { spec { assert false; }; abort 10 };
            i = i + 1;
        }
    }
    spec fun iter10_no_abort { // FIXME: Should be proved.
        pragma verify=true;
        //aborts_if false;
    }
```